ledger-service/lf-value-json/src/main/scala/com/digitalasset/daml/lf/value/json/ApiCodecCompressed.scala

```diff
@@ -136,7 +136,13 @@ class ApiCodecCompressed[Cid](val encodeDecimalAsString: Boolean, val encodeInt6
       case Model.DamlLfPrimType.Timestamp => { case JsString(v) =>
         V.ValueTimestamp(assertDE(Time.Timestamp fromString v))
       }
-      case Model.DamlLfPrimType.Date => { case JsString(v) => V.ValueDate.fromIso8601(v) }
+      case Model.DamlLfPrimType.Date => { case JsString(v) =>
+        try {
+          V.ValueDate.fromIso8601(v)
```

Comment:
probably not in scope for this PR but we might want to consider having `fromIso8601`return an `Either` and having an `assertFromIso8601` that throws like we do in lots of other places.

Reply:
@realvictorprm Since you offered, do you want to take care of this part?

Reply:
Will do :)

Reply:
Thanks!